test/widgets/subscription_list_test.dart

```diff
@@ -186,6 +187,49 @@ void main() {
     check(find.byType(UnreadCountBadge).evaluate()).length.equals(0);
   });
 
+  testWidgets('muted unread badge shows with muted unreads', (tester) async {
```

Comment:
The structure of these test cases looks fine.

This case should have the opposite expectation, though — the topic (and not only the stream/channel) is muted, so there should be no badge.

Then let's also have test cases exercising each of the possible scenarios:
* all the possible combinations of isTopicVisible true/false and isTopicVisibleInStream true/false;
* for each of those, all the possible values for whether the stream is muted.

This muted-unreads badge is closely related to the unread-count badge. So some of those scenarios may be covered by the existing test cases above, and those can be handled by just adding another check to those cases.